inline-r/tests/Test/Regions.hs

```diff
@@ -43,35 +40,47 @@ assertBalancedStack m = do
 
 tests :: TestTree
 tests = testGroup "regions"
-    [ testCase "qq-dont-leak" $
-      preserveDirectory $ assertBalancedStack $
+    [ testCase "qq-object-live-inside-extend" $
+      assertBalancedStack $
         runRegion $ do
-          _ <- [r| gctorture(TRUE) |]
           R.SomeSEXP x <- [r| 1 |]
-          _ <- io $ R.allocList 1
+          _ <- [r| gc () |]
           io $ assertEqual "value is protected" R.Real (R.typeOf x)
-          _ <- [r| gctorture(FALSE) |]
-          return ()
-    , testCase "mksexp-dont-leak" $
-      preserveDirectory $ assertBalancedStack $
+    , testCase "mksexp-object-live-inside-extend" $
+      assertBalancedStack $
         runRegion $ do
-          _ <- [r| gctorture(TRUE) |]
           x <- mkSEXP (1::Int32)
-          _ <- io $ R.allocList 1
+          _ <- [r| gc () |]
```

Comment:
I'm not sure this change is correct in general case. With simple `gc` R will mark object as unused but without `--enable-strict-barrier` object will not be changed. As a result  next assertion will pass, even value was deallocated. (This is not a case with `--enable-strict-barrier` but our test suite should not rely on proper configured R).
Previous solution worked even without `--enable-strict-barrier` enabled, as we were allocating an int, and on list allocation it was guaranteed that list will be allocated in the place where `Int32` was. As a result assertion will fail if value was not protected.